src/main/java/com/fasterxml/jackson/databind/deser/std/CollectionDeserializer.java

```diff
@@ -341,12 +341,10 @@ protected Collection<Object> _deserializeFromArray(JsonParser p, Deserialization
         // [databind#631]: Assign current value, to be accessible by custom serializers
         p.assignCurrentValue(result);
 
-        JsonDeserializer<Object> valueDes = _valueDeserializer;
```

Comment:
If it's ok, I'll merge these into 2.19 separately to reduce diff? And can then merge to PR.

Reply:
Done via #5141, 2.19 updated.

Reply:
Merged into PR to reduce diff.